### PR TITLE
TIP-820: Replace deprecated assetic bundle with less.js

### PR DESCRIPTION
**Pull request by tamarasaurus**
**Description (for Contributor and Core Developer)**

This PR removes the deprecated assetic bundle and replaces it with less.js to compile our .less files.

In the package.json we now have two new scripts:

```bash
# Uses less.js to compile the .less files from each required bundle and outputs the css to web/css/pim.css
yarn run less

# Watches for changes in the EE and CE .less and css files and executes 'yarn run less'
yarn run less-watch
```

The old command `bin/console oro:assetic:dump` is replaced by `yarn run less`

The `assets.yml` file in each bundle is replaced with a `Resources/public/less/index.less` file that imports the required stylesheets. To migrate assets.yml file:

Before
```yml
css:
    lib:
        - bundles/somebundle/assets/less/wysiwyg.css
        - bundles/somebundle/assets/less/permission.less
```

After
```less
@import (less) "./web/bundles/somebundle/assets/less/wysiwyg.css";
@import "./web/bundles/somebundle/assets/less/permission.less";
```

The old Oro Assetic bundle had some behaviour to rewrite the urls in stylesheets, this is replaced by a lessjs plugin here - https://github.com/akeneo/pim-community-dev/blob/replace-assetic/frontend/build/less-rewrite-urls.js

Changes
- [x] Update the CI and installation scripts
- [x] Remove the assetic bundle
- [x] Refactor the rewrite url plugin and compile-less script
- [x] Add file watcher and rebuild
- [x] Document upgrade migration
- [x] Add minification for prod
- [x] Output source map depending on env
- [ ] Standard PRs
- [ ] PIM docs PR
- [ ] Ansible PR

Rebuild on file changes + live reload in the browser in under 1 second
test

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | OK
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -
 
`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -41,6 +41,9 @@ jobs:
       - run:
           name: Pim database installation
           command: docker-compose exec fpm bin/console --env=behat pim:install --force --clean
+      - run:
+            name: Pim yarn less installation
```
Comment (tamarasaurus) [inline .circleci/config.yml]: Fix alignment

File: UPGRADE.md [hunk 2]
```diff
@@ -260,10 +260,34 @@ Before updating the dependencies and migrating your data, please deactivate all
...
+    ```
+
+    After in `Resources/public/less/index.less`
+
+    ```less
+        @import (less) "./web/bundles/yourbundle/assets/less/styles.css";
+        @import "./web/bundles/yourbundle/assets/less/bundle.less";
+    ```
+
+    If you are importing a .css file, you must add `(less)` after the import, as above. If you only have .less files in your bundle's assets.yml, you can remove it.
```
Comment (pierallard) [inline UPGRADE.md]: That's weird, no?

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/less/index.less [hunk 3]
```diff
@@ -0,0 +1,112 @@
+
```
Comment (pierallard) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/less/index.less]: Space

File: UPGRADE.md (hunk 2 above)
Comment (tamarasaurus) [reply UPGRADE.md]: Yeah, by default less.js won't parse the import if the file extension is .css, it just leaves the '@import' statement in the code - http://lesscss.org/features/#import-atrules-feature-import-options so we have to do that

**Pull request merged**